app/frontend/src/submissions/components/SubmissionForm/Owner.vue

```diff
@@ -135,14 +145,108 @@ export default {
     ownerTelInput: 'ownerTel'
   },
   data () {
-    return {}
+    return {
+      addressSuggestions: [],
+      isLoadingSuggestions: false
+    }
   },
   computed: {
     ...mapGetters(['codes'])
+  },
+  methods: {
+    /**
+     * Asynchronously fetches address suggestions based on the owner's address input.
+     * If no input is provided, it clears the current suggestions.
+     * On success, it maps the received data to full addresses and updates the addressSuggestions state.
+     * On failure, it logs the error and clears the current suggestions.
+     * Finally, sets the loading state to false.
+     */
+    async fetchAddressSuggestions() {
+      if (!this.ownerAddressInput) {
+        this.addressSuggestions = [];
+        return;
+      }
+      this.isLoadingSuggestions = true;
+      const params = {
+        minScore: 50, //accuracy score of results compared to input
+		    maxResults: 5,
+        echo: 'false',
+        brief: true,
+        autoComplete: true,
+        addressString: this.ownerAddressInput
+      };
+
+      const querystring = require('querystring');
+      const searchParams = querystring.stringify(params);
+      try {
+        const response = await fetch(`${GEOCODER_ADDRESS_API}${searchParams}`);
+        const data = await response.json();
+        if (data && data.features) {
+          
+          this.addressSuggestions = data.features.map(item => item.properties.fullAddress);
+        } else {
+          this.addressSuggestions = [];
+        }
+      } catch (error) {
+        console.error(error);
+        this.addressSuggestions = [];
+      } finally {
+        this.isLoadingSuggestions = false;
+      }
+    },
+
+    /**
+     * Processes the selected address suggestion.
+     * Splits the suggestion into components and updates the owner's province, city, and address inputs accordingly.
+     * Clears the address suggestions afterward.
+     * @param {string} suggestion - The selected address suggestion. ("1234 Street Rd, Name of City, BC")
+     */
+    selectAddressSuggestion(suggestion) {
+      const ownerAddressArray = suggestion.split(',');
+      if(ownerAddressArray){
+        if(ownerAddressArray[ownerAddressArray.length -1].toUpperCase().trim() === 'BC' || ownerAddressArray[-1].toUpperCase().trim() === 'BRITISH COLUMBIA')
+        {
+          this.ownerProvinceInput = this.codes.province_codes[0].province_state_code;
+          this.ownerAddressInput = '';
+        }
+      else this.ownerProvinceInput = "";
+      this.ownerCityInput = ownerAddressArray[ownerAddressArray.length -2];
+      if(ownerAddressArray[ownerAddressArray.length -3]) this.ownerAddressInput = ownerAddressArray[ownerAddressArray.length -3];
```

Comment:
For readability/maintainability you should declare some variables in the function since you're using -1, -2, -3 to get

```js
const namedValue1 = ownerAddressArray.length - 1;
const namedValue2 = ownerAddressArray.length - 2;
const namedValue3 = ownerAddressArray.length - 3;
```

Reply:
corrected